src/main.c

```diff
@@ -37,15 +37,15 @@ static char *read_file(const char *path, size_t *size) {
 static void dump_ast(const char *code, int binary) {
   struct ast ast;
 
-  ast_init(&ast, 0);
+  mbuf_init(&ast.mbuf, 0);
```

Comment:
Makes sense. That will keep AST implementation private to AST.
I've changed ast_trim() to ast_optimize().
PTAL.